src/streaming.ts

```diff
@@ -46,46 +52,116 @@ export class StreamProcessor {
     this.eventBus = eventBus;
     this.repository = repository;
     this.log = this.options.logger;
+
+    const minDelay = 5000;
+    const maxDelay = 10000;
+    this.retryAttempt = 1;
+    this.retryDelayMs = Math.floor(
+      Math.random() * (maxDelay - minDelay + 1) + minDelay,
+    );
```

Comment:
I'm confused by this... since `minDelay` and `maxDelay` are both consts, then this equates to `Math.random() * 5001 + 5000` is that intentional? Seems pretty arbitrary and will always come up with a number between 5.001s and 10.001s.

Reply:
yeah it increments a random delay between 5 and 10 seconds up until a max of 60 seconds. 5-10 seconds was chosen to prevent it reaching 60s too fast. not the most sophisticated back off strategy but simple to implement with some level of randomness. the expression is a bit confusing because i couldn't find a method to give me a random number between two ranges in node.

Reply:
I know what you mean, I miss `rand(min, max)` in PHP. But anyway, you can safely remove the `+1` part and it'll be fine unless you really want it to be 5.001s and 10.001s